packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -742,7 +742,7 @@ class _CupertinoSliverNavigationBarState extends State<CupertinoSliverNavigation
           transitionBetweenRoutes: widget.transitionBetweenRoutes,
           heroTag: widget.heroTag,
           persistentHeight: _kNavBarPersistentHeight + MediaQuery.of(context).padding.top,
-          alwaysShowMiddle: widget.middle != null,
+          alwaysShowMiddle: false,
```

Comment:
If this is always going to be false, can this field be removed and the logic further down updated to not be conditional? Also the comment on line 803 would need to be updated to avoid confusion later.